### PR TITLE
Add commands to get preset names

### PR DESCRIPTION
**Pull request by benmcmorran**
### This changes commands for command substitution and associated documentation.

The following changes are proposed:

3 new commands: `cmake.activeConfigurePresetName`, `cmake.activeBuildPresetName`, `cmake.activeTestPresetName`. Each returns a string with the name of the corresponding preset, or an empty string if the preset is not available.

## The purpose of this change

These commands are designed to be used with the command substitution syntax in `launch.json` or `tasks.json`. They help to write generic launch configurations or tasks in the presence of CMake projects with multiple presets that may use the `${presetName}` macro to select the `binaryDir`. While the existing `cmake.launchTargetPath` can often also be used in this situation, it can't handle constructing the path to files that aren't the target (e.g. associated debug information or, for embedded applications, hex or bin files).

FYI @aleun @gcampbell-msft

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Improvements:
 - Allow editing Kits when presets are in use. [#1965](https://github.com/microsoft/vscode-cmake-tools/issues/1965)
 - Launch the target in the default terminal. [PR #2311](https://github.com/microsoft/vscode-cmake-tools/pull/2311) [@michallukowski](https://github.com/michallukowski)
 - Allow launching targets in parallel. [#2240](https://github.com/microsoft/vscode-cmake-tools/issues/2120) [@ColinDuquesnoy](https://github.com/ColinDuquesnoy)
+- Add commands to get preset names. [#2433](https://github.com/microsoft/vscode-cmake-tools/pull/2433)
```
Comment (benmcmorran) [inline CHANGELOG.md:11]: As Microsoft FTE I left my name out, but if you want me to add it I can.

Comment (bobbrow) [reply CHANGELOG.md:11]: Good question.  It is intentional that we only list non-MSFT folks in our changelog.

Comment (elahehrashedi) [reply CHANGELOG.md:22]: please modify [#2433] to [PR #2433] and move it below 1.11 before merging.

**Review by elahehrashedi — APPROVED** [review]

File: docs/cmake-settings.md [hunk 2]
```diff
@@ -89,6 +89,9 @@ Supported commands for substitution:
 |`cmake.tasksBuildCommand`|The CMake command used to build your project based on the currently selected Kit + Variant + Target. Suitable for use within `tasks.json`.|
 |`cmake.activeFolderName`|The name of the active folder (e.g. in a multi-root workspace)|
 |`cmake.activeFolderPath`|The asolute path of the active folder (e.g. in a multi-root workspace)|
+|`cmake.activeConfigurePresetName`|The name of the selected configure preset.|
```
Comment (elahehrashedi) [inline docs/cmake-settings.md:92]: Please replace "selected" with "active" to be consistent with the other descriptions.

**Comment by elahehrashedi** [comment]
@benmcmorran there has been some renaming recently, can you apply it to your branch so that the builds pass, they are failing currently.

**Pull request merged**